delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/OrganizationsDummy.scala

```diff
@@ -32,15 +33,19 @@ final class OrganizationsDummy private (
     journal: OrganizationsJournal,
     cache: OrganizationsCache,
     semaphore: IOSemaphore,
-    applyOwnerPermissions: ApplyOwnerPermissionsDummy
+    scopeInitializations: Set[ScopeInitialization]
```

Comment:
Organizations know too much, it should not know that there are ScopeInitialization behind the scene or decide if it should run in parallel or not.
From its point of view it is just a function that takes the fresh org and return an `IO[OrganizationInitializationFailed, Unit]` (and it should make testing easier).
Maybe this how we run them should be delegated to a ScopeInitializationRunner or something ?

Reply:
I think introducing a Runner makes things more complicated... I don't see any benefit for adding an arbitrary function that gets invoked. The behaviour is explicit: organizations and projects have an initialization hook allows for arbitrary behaviour to be registered as per construction. Unless we would like to prioritize the execution of certain initialization behaviours (no usecase for that so far), I see the abstraction adding some unnecessary complexity.

Reply:
To ensure we run it the same way in synchronous mode and healing mode and to make tests easier without having to worry what a ScopeInitialization is ?

Reply:
Given the fact that there are only 2 places where this code is used, the customization of the tracing span and aligning errors I think it would be too much to generalize/abstract.